apps/docs/components/ProjectConfigVariables.tsx

```diff
@@ -0,0 +1,455 @@
+function fromStoredString(maybeInstanceKey: string) {
+  try {
+    return JSON.parse(maybeInstanceKey)
+  } catch {
+    return null
+  }
+}
+
+function escapeSpacedSlashes(str: string) {
+  return str.replace(/ \/ /g, ' [forward-slash] ')
+}
+
+function restoreSpacedSlashes(str: string) {
+  return str.replace(/ \[forward-slash\] /g, ' / ')
+}
+
+function toPrettyString(instance: InstanceData | null) {
+  if (!instance) return ''
+
+  const { organization, project, branch = null } = instance
+  return `${
+    organization ? `${escapeSpacedSlashes(organization.name)} / ` : ''
+  }${escapeSpacedSlashes(project.name)}${branch ? ` / ${escapeSpacedSlashes(branch.name)}` : ''}`
+}
+
+function fromPrettyString(instances: InstanceData[], maybeInstance: string) {
+  const [organizationNameEscaped, projectNameEscaped, branchNameEscaped] =
+    maybeInstance.split(' / ')
+  const organizationName = organizationNameEscaped
+    ? restoreSpacedSlashes(organizationNameEscaped)
+    : null
+  const projectName = projectNameEscaped ? restoreSpacedSlashes(projectNameEscaped) : null
+  const branchName = branchNameEscaped ? restoreSpacedSlashes(branchNameEscaped) : null
+
+  if (organizationName && projectName) {
+    return (
+      instances.find(
+        (instance) =>
+          // This gets filtered through `Command`'s `value` prop,
+          // which automatically converts to lowercase
+          instance.organization?.name.toLowerCase() === organizationName &&
+          instance.project?.name.toLowerCase() === projectName &&
+          (!instance.branch || instance.branch.name.toLowerCase() === branchName)
+      ) ?? null
+    )
+  }
+
+  return null
+}
+
+const instanceStore = proxy({
+  selectedInstanceKey: null as InstanceKey | null,
+  setSelectedInstanceKey: (instanceKey: InstanceKey | null) => {
+    instanceStore.selectedInstanceKey = instanceKey
+    if (instanceKey !== null) {
+      store('local', LOCAL_STORAGE_KEYS.SAVED_ORG_PROJECT_BRANCH, toStoredString(instanceKey))
+    }
+  },
+  clear: () => {
+    instanceStore.setSelectedInstanceKey(null)
+  },
+})
+
+type Variable = 'url' | 'anonKey'
+
+const prettyFormatVariable: Record<Variable, string> = {
+  url: 'Project URL',
+  anonKey: 'Anon key',
+}
+
+type ProjectConfigVariablesState =
+  | 'userLoading'
+  | 'loggedOut'
+  | 'loggedIn.dataPending'
+  | 'loggedIn.hasData'
+  | 'loggedIn.hasNoData'
+  | 'loggedIn.dataError'
+
+type ProjectConfigVariablesPlusApiState =
+  | Omit<ProjectConfigVariablesState, 'loggedIn.hasData'>
+  | 'loggedIn.hasData.apiDataPending'
+  | 'loggedIn.hasData.apiDataSuccess'
+  | 'loggedIn.hasData.apiDataError'
+
+function ComboBox({
+  parentStateSummary,
+  instances,
+  className,
+}: {
+  parentStateSummary: ProjectConfigVariablesState
+  instances: InstanceData[]
+  className?: string
+}) {
+  const { selectedInstanceKey, setSelectedInstanceKey } = useSnapshot(instanceStore)
+  const selectedInstance = getInstanceDataFromKey(instances, selectedInstanceKey)
```

Comment:
it does look like we're showing branch instances here (I'm guessing those with 3 breadcrumbs) - is that what we want?

![image](https://github.com/supabase/supabase/assets/19742402/62dad44a-18cb-4328-9bbf-4c28c02e2256)

Reply:
yeah thinking twice - this makes sense 👍